pyclibrary/c_parser.py

```diff
@@ -35,6 +35,91 @@
 __all__ = ['win_defs', 'CParser']
 
 
+def wrap_int(t):
+    logger.debug('wrap_int: {} {}'.format(t.dump(), type(t)))
+    t[0] = "CInt({})".format(eval(t[0]))
```

Comment:
Please rewrite without using eval.

Reply:
hello @kalvdans,

Thanks for your reply, avoiding using eval is a very good simple suggestion to avoid security issues and everyone should know.

One solution is to write a C integer parser, but I'm not sure it's worth writing. Because my purpose is to bind existing open source c libraries. If some people who want to use untrusted libraries may be worried about this.

Also, I think must write in clean code for better maintenance, but I don't know of any python built-in function that can convert C integer string to a Python integer. If you know a function, I will appreciate if you tell me :)

See also this code https://github.com/r888800009/pyclibrary/blob/f5f8bf80113b9deb247e92705bb4b73529b95912/pyclibrary/c_parser.py#L1736-L1739

The token passed in by wrapper must match the regular expression. If you know how to bypass this regex to achieve python execute code, please tell me and I will be able to confirm that this is indeed a bad approach

Of course, to avoid misuse, we'd better avoiding using eval, but we also need have a good enough solution to make it easy to maintain.

For your concerns, maybe you'll want to look at the existing code in c_parser.py

https://github.com/MatthieuDartiailh/pyclibrary/blob/1d4dbfc207afee3fd80b72e94c60d47d8263d49a/pyclibrary/c_parser.py#L1514

If we care about security issues, maybe we can consider rewriting the eval function of the entire project

Reply:
I did not check in depth but could you use ast.literal_eval ?

Reply:
hmm, `int(t[0], 0)` should handle 0x prefixes fine, but doesn't work with octal.

Reply:
It seems we can simply write an octal parser https://github.com/MatthieuDartiailh/pyclibrary/commit/44daba6f06701d1e56e721a99e49a2e62164c5fb to solve this problem without too much complexity.

new patch https://github.com/MatthieuDartiailh/pyclibrary/pull/66/commits/a7f7624734a107e0438a8ec36d339e43fbcc61b7  uses `int(t[0], 0)`, which can restrict the input of only integer.